pkg/controller/eventing/inmemory/channel/reconcile.go

```diff
@@ -0,0 +1,466 @@
+// OwnerReferences on the resource so handleObject can discover the Channel resource that 'owns' it.
+// As well as being garbage collected when the Channel is deleted.
+func newK8sService(c *eventingv1alpha1.Channel) *corev1.Service {
+	labels := map[string]string{
+		"channel":     c.Name,
+		"provisioner": c.Spec.Provisioner.Ref.Name,
+	}
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      controller.ChannelServiceName(c.ObjectMeta.Name),
+			Namespace: c.Namespace,
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(c, schema.GroupVersionKind{
+					Group:   eventingv1alpha1.SchemeGroupVersion.Group,
+					Version: eventingv1alpha1.SchemeGroupVersion.Version,
+					Kind:    "Channel",
+				}),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name: PortName,
+					Port: PortNumber,
+				},
+			},
+		},
+	}
+}
+
+// newVirtualService creates a new VirtualService for a Channel resource. It also sets the
+// appropriate OwnerReferences on the resource so handleObject can discover the Channel resource
+// that 'owns' it. As well as being garbage collected when the Channel is deleted.
+func newVirtualService(channel *eventingv1alpha1.Channel) *istiov1alpha3.VirtualService {
+	labels := map[string]string{
+		"channel":     channel.Name,
+		"provisioner": channel.Spec.Provisioner.Ref.Name,
+	}
+	destinationHost := controller.ServiceHostName(controller.ClusterBusDispatcherServiceName(channel.Spec.Provisioner.Ref.Name), system.Namespace)
+	return &istiov1alpha3.VirtualService{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      controller.ChannelVirtualServiceName(channel.Name),
+			Namespace: channel.Namespace,
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(channel, schema.GroupVersionKind{
+					Group:   eventingv1alpha1.SchemeGroupVersion.Group,
+					Version: eventingv1alpha1.SchemeGroupVersion.Version,
+					Kind:    "Channel",
+				}),
+			},
+		},
+		Spec: istiov1alpha3.VirtualServiceSpec{
+			Hosts: []string{
+				controller.ServiceHostName(controller.ChannelServiceName(channel.Name), channel.Namespace),
+				controller.ChannelHostName(channel.Name, channel.Namespace),
+			},
+			Http: []istiov1alpha3.HTTPRoute{{
+				Rewrite: &istiov1alpha3.HTTPRewrite{
+					Authority: controller.ChannelHostName(channel.Name, channel.Namespace),
+				},
+				Route: []istiov1alpha3.DestinationWeight{{
+					Destination: istiov1alpha3.Destination{
+						Host: destinationHost,
+						Port: istiov1alpha3.PortSelector{
+							Number: PortNumber,
+						},
+					}},
+				}},
+			},
+		},
+	}
+}
+
+func (r *reconciler) setStatusReady(c *eventingv1alpha1.Channel) {
+	c.Status.Conditions = []duckv1alpha1.Condition{
+		{
+			Type:   duckv1alpha1.ConditionReady,
+			Status: corev1.ConditionTrue,
+		},
+	}
+}
+func (r *reconciler) updateChannel(ctx context.Context, u *eventingv1alpha1.Channel) error {
```

Comment:
I don't think I agree. At least the current Subscription controller seems to handle it in the reconciler, rather than in Subscription itself. [diffing](https://github.com/knative/eventing/blob/7baba63f9488e8937ab704d79e6a96984a9fe4b9/pkg/controller/eventing/subscription/reconcile.go#L62-L70) + [updating](https://github.com/knative/eventing/blob/7baba63f9488e8937ab704d79e6a96984a9fe4b9/pkg/controller/eventing/subscription/reconcile.go#L143).

Reply:
Ping @n3wscott